substrate/advanced/io

```diff
@@ -26,72 +25,72 @@ namespace substrate
 			SUBSTRATE_NO_DISCARD(virtual ssize_t read(void *bufferPtr,
 				size_t bufferLen, std::nullptr_t) const noexcept) = 0;
 
-			SUBSTRATE_NO_DISCARD(bool read(void *const value, const size_t valueLen, size_t &resultLen) const noexcept)
+			SUBSTRATE_NO_DISCARD(inline bool read(void *const value, const size_t valueLen, size_t &resultLen) const noexcept)
 			{
-				const ssize_t result = read(value, valueLen, nullptr);
+				const ssize_t result{read(value, valueLen, nullptr)};
 				if (result < 0)
 					return false;
 				resultLen = size_t(result);
 				return resultLen == valueLen;
 			}
 
-			SUBSTRATE_NO_DISCARD(bool read(void *const value, const size_t valueLen) const noexcept)
+			SUBSTRATE_NO_DISCARD(inline bool read(void *const value, const size_t valueLen) const noexcept)
 			{
-				size_t resultLen = 0;
+				size_t resultLen{};
 				return read(value, valueLen, resultLen);
 			}
 
-			template<typename T> bool read(T &value) const noexcept
+			template<typename T> inline bool read(T &value) const noexcept
```

Comment:
Where possible, it would be nice as a library enhancement to mark these all `SUBSTRATE_NO_DISCARD()` as these should all be checked returns ideally